inference-engine/src/vpu/graph_transformer/src/stages/activation.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vpu/frontend/frontend.hpp>
+
+using namespace InferenceEngine;
+
+namespace vpu {
+
+void FrontEnd::parseLogicalNot(const Model &model, const ie::CNNLayerPtr &layer, const DataVector &inputs, const DataVector &outputs) const {
+    LayerParams params = {layer->name, "Eltwise", layer->precision};
+    auto res = std::make_shared<InferenceEngine::EltwiseLayer>(params);
+    res->_operation = InferenceEngine::EltwiseLayer::Logical_NOT;
+
+    parseEltwise(model, res, inputs, outputs);
+}
+
+void FrontEnd::parseActivation(const Model& model, const ie::CNNLayerPtr& layer, const DataVector& inputs, const DataVector& outputs) const {
+    const ie::details::caseless_map<std::string, LayerParser> activationParsers {
+        {"not", LAYER_PARSER(parseLogicalNot)},
+    };
+
+    auto type = layer->GetParamAsString("type");
+
+    VPU_THROW_UNSUPPORTED_UNLESS(activationParsers.find(type) != activationParsers.end(),
+                                 "Failed to compile layer \"%v\"(type = %v) ", layer->name, type);
+
+    activationParsers.at(type)(model, layer, inputs, outputs);
```

Comment:
`activationParsers.find()` and `activationParsers.at()` are double-searching through map. I understand, map contains 1 element, but there is really no need to do this

Reply:
Do you mean we don't need to catch an exception at all?

Reply:
I mean something like this:
```
const auto activationParserIt = activationParsers.find(type);
VPU_THROW_UNSUPPORTED_UNLESS(activationParserIt != activationParsers.end(), ...);
activationParserIt->second(model, layer, inputs, outputs);
```